apps/qa/src/pages/ingest/ingest.py

```diff
@@ -2,20 +2,41 @@ def ingest():
     import streamlit as st
     from .helpers import archive_raw_data
     from pathlib import Path
+    import time
 
     st.title("Ingest Dataset")
 
-    dataset_name = st.text_input("Dataset Name")
-    version = st.text_input("Version")
-    uploaded_file = st.file_uploader("Choose a file")
+    if "running" not in st.session_state:
+        st.session_state.running = False
+
+    dataset_name = st.text_input("Dataset Name", disabled=st.session_state.running)
+    version = st.text_input("Version", disabled=st.session_state.running)
+    uploaded_file = st.file_uploader("Choose a file", disabled=st.session_state.running)
     s3_path = Path("inbox") / dataset_name / version
 
-    ingest_button_pressed = st.button("Ingest")
+    disabled = st.session_state.get("running", False)
+    ingest_button_pressed = st.button("Ingest", disabled=st.session_state.running)
     if ingest_button_pressed == True:
         if dataset_name and version and uploaded_file:
-            file_name = uploaded_file.name
-            file_path = archive_raw_data(
-                dataset_name, version, uploaded_file, file_name
-            )
+            st.session_state.running = True
+            st.rerun()
         else:
             st.warning("Please input all fields.")
+
+    if st.session_state.running == True:
```

Comment:
For this (and other cases), you can drop the ` == True`! Just `if st.session_state["running"]`